be/src/agent/heartbeat_server.cpp

```diff
@@ -150,6 +152,20 @@ void HeartbeatServer::heartbeat(THeartbeatResult& heartbeat_result, const TMaste
     }
 }
 
+std::string HeartbeatServer::print_master_info(const TMasterInfo& master_info) const {
+    MasterInfo info(master_info);
+
+    if (info.__isset.token) {
+        info.__set_token = "<hidden>";
+    }
+
+    std::ostringstream out;
+
+    master_info_copy.printTo(out);
+
+    return out.str();
```

Comment:
```suggestion
    std::ostringstream out;
    if (!master_info.__isset.token) {
        master_info.printTo(out);
    } else {
        MasterInfo info(master_info);
        info.__set_token = "<hidden>";
        info.printTo(out);
    }
    return out.str();
```

Save the object clone if the token is not set.